R/carehomes.R

```diff
@@ -524,6 +455,54 @@ carehomes_parameters <- function(start_date, region,
                                                   strain$n_strains,
                                                   vaccine_catchup_fraction)
 
+
+  strain_rel_severity <- recycle(assert_relatives(strain_rel_severity),
+                                 length(strain_transmission))
+  if (length(strain_transmission) > 1) {
+    strain_rel_severity <- mirror_strain(strain_rel_severity)
+  }
+  rel_severity <- aperm(array(strain_rel_severity,
+                              c(strain$n_strains, ret$n_groups,
+                                vaccination$n_vacc_classes)),
+                        c(2, 1, 3))
+  ret$rel_p_ICU <- array(1, c(ret$n_groups, strain$n_strains,
+                              vaccination$n_vacc_classes))
+  ret$rel_p_ICU_D <- rel_severity
+  ret$rel_p_H_D <- rel_severity
+  ret$rel_p_W_D <- rel_severity
+  ret$rel_p_G_D <- rel_severity
+
+
+  strain_rel_gamma_A <- recycle(assert_relatives(strain_rel_gamma_A),
+                                length(strain_transmission))
+  strain_rel_gamma_P <- recycle(assert_relatives(strain_rel_gamma_P),
+                                length(strain_transmission))
+  strain_rel_gamma_C_1 <- recycle(assert_relatives(strain_rel_gamma_C_1),
+                                  length(strain_transmission))
+  strain_rel_gamma_C_2 <- recycle(assert_relatives(strain_rel_gamma_C_2),
+                                  length(strain_transmission))
+  ret$rel_gamma_E <- rep(1, strain$n_strains)
+  if (length(strain_rel_gamma_A) == 2) {
+    ret$rel_gamma_A <- mirror_strain(strain_rel_gamma_A)
+    ret$rel_gamma_P <- mirror_strain(strain_rel_gamma_P)
+    ret$rel_gamma_C_1 <- mirror_strain(strain_rel_gamma_C_1)
+    ret$rel_gamma_C_2 <- mirror_strain(strain_rel_gamma_C_2)
+  } else {
+    ret$rel_gamma_A <- rep(1, strain$n_strains)
+    ret$rel_gamma_P <- rep(1, strain$n_strains)
+    ret$rel_gamma_C_1 <- rep(1, strain$n_strains)
+    ret$rel_gamma_C_2 <- rep(1, strain$n_strains)
+  }
```

Comment:
I wonder if we could simplify this slightly, something like (I have not tested this locally, also Rich doesn't like `get` so maybe not the best idea)

```suggestion
strain_ones <- rep(1, strain$n_strains)
for (str in c("rel_gamma_A", "rel_gamma_P", "rel_gamma_C_1", "rel_gamma_C_2")) {
  gamma <- get(sprintf("strain_%s", str))
  gamma <- recycle(assert_relatives(gamma), length(strain_transmission))
  if (length(gamma) == 2) {
    ret[[str]] <- mirror_strain(gamma)
  } else {
    ret[[str]] <- strain_ones 
  }
  ret$rel_gamma_E <- strain_ones
}
```

Reply:
Have done this but in a way that avoids using `get`